lib/inspec/profile.rb

```diff
@@ -89,7 +89,7 @@ def initialize(source_reader, options = {})
       @writable = options[:writable] || false
       @profile_id = options[:id]
       @cache = options[:cache] || Cache.new
-      @backend = options[:backend] || Inspec::Backend.create(options)
+      @backend = options[:backend] || Inspec::Backend.create(options.select { |k, _| k != 'target' })
```

Comment:
To me this area of the code is a bit opaque (regarding passing of a `:target`) without knowing what the message spec from Profile -> Backend should look like under different circumstances.  I agree that we should determine what the message(s) should contain and be able to enforce that on the receiving end.